packages/gatsby-source-drupal/src/gatsby-node.js

```diff
@@ -176,13 +176,21 @@ exports.onCreateDevServer = (
     bodyParser.text({
       type: `application/json`,
     }),
-    async (req, res) => {
-      // we are missing handling of node deletion
-      const nodeToUpdate = JSON.parse(JSON.parse(req.body)).data
-
-      await handleWebhookUpdate(
+    async (req, _res) => {
+      const requestBody = JSON.parse(JSON.parse(req.body))
+      const { data, secret, action, id } = requestBody
+      if (pluginOptions.secret && pluginOptions.secret !== secret) {
+        return reporter.warn(
+          `The secret in this request did not match your plugin options secret.`
+        )
+      }
+      if (action === `delete`) {
+        actions.deleteNode({ node: getNode(createNodeId(id)) })
```

Comment:
I do have question here:
What happens if given node is being referenced (relationship) by other node? Does drupal automatically update other posts that reference deleted content and remove those? Or are those not touched and then there is content that still references deleted content?

We definitely will need to add some code here to handle back references (I can work on that), question is if we need to handle normal references as well or drupal actually updates those automatically

Reply:
That is a great question. Ive asked for some insight into this on drupal, but I think a modified/extended version of what we already use to update relationships should be able to handle this.

Is this something `deleteNode` itself should/could handle?

Reply:
This is something that gatsby core potentially could handle, but doesn't do it right now (it's up to plugins to keep track of relationships - mostly because it's usually more efficient than any generic solution we could have in core)

Reply:
It seems from my conversation with someone who knows Drupal, the relationship are not necessarily updated on delete and it depends on how content and relationships are set up as to when and how that update or removal may happen. ugh.

Reply:
here is an issue discussing the problem https://www.drupal.org/project/drupal/issues/2723323
Some of the fixes purely deal with Drupal view which would solve for JSON API problems.